### PR TITLE
allow polaris to run in foreground

### PR DESCRIPTION
**Pull request by jxs**
allow polaris to run in foreground. This is very helpful if you want to launch it via Systemd or any other init system, and then get its status directly via stdout/sdterr and also let the init system manage logs

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -100,17 +100,28 @@ fn daemonize() -> Result<()> {
...
-	#[cfg(unix)]
-	daemonize()?;
-
 	// Parse CLI options
 	let args: Vec<String> = std::env::args().collect();
 	let mut options = Options::new();
 	options.optopt("c", "config", "set the configuration file", "FILE");
 	options.optopt("d", "database", "set the path to index database", "FILE");
 	options.optopt("w", "web", "set the path to web client files", "DIRECTORY");
+	options.optflag("f", "foreground", "run polaris in the foreground instead of daemonizing");
```
Comment (agersant) [inline src/main.rs]: Polaris never runs as a daemon on Windows, can you add the #[cfg(unix)] conditional compilation flag here?

File: src/main.rs [hunk 2]
```diff
@@ -100,17 +100,28 @@ fn daemonize() -> Result<()> {
...
 
+	if matches.opt_present("h") {
+		let program = args[0].clone();
+		let brief = format!("Usage: {} [options]", program);
+		print!("{}", options.usage(&brief));
+		return Ok(());
+	}
+
+	if !matches.opt_present("f") {
+		daemonize()?;
```
Comment (agersant) [inline src/main.rs]: Same conditional compilation needed here, daemonize doesn't exist on Windows.

**Comment by jxs** [comment]
yes of course! updated :)
I somehow thought windows was not using cli flags at all

**Comment by agersant** [comment]
Thank you!

**Pull request merged**